src/platform/crypt_openssl.c

```diff
@@ -27,20 +27,102 @@
 #include "openssl/hmac.h"
 #endif
 #include "openssl/err.h"
+#include "openssl/evp.h"
 #include "openssl/kdf.h"
 #include "openssl/pem.h"
 #include "openssl/pkcs12.h"
 #include "openssl/pkcs7.h"
 #include "openssl/rsa.h"
 #include "openssl/ssl.h"
 #include "openssl/x509.h"
+
 #ifdef _WIN32
 #pragma warning(pop)
 #endif
 #ifdef QUIC_CLOG
 #include "crypt_openssl.c.clog.h"
 #endif
 
+EVP_CIPHER *CXPLAT_AES_128_GCM_ALG_HANDLE;
+EVP_CIPHER *CXPLAT_AES_256_GCM_ALG_HANDLE;
+EVP_CIPHER *CXPLAT_AES_128_ECB_ALG_HANDLE;
+EVP_CIPHER *CXPLAT_AES_256_ECB_ALG_HANDLE;
+EVP_CIPHER *CXPLAT_CHACHA20_POLY1305_ALG_HANDLE;
+#ifdef IS_OPENSSL_3
+EVP_MAC_CTX *CXPLAT_HMAC_SHA256_CTX_HANDLE;
+EVP_MAC_CTX *CXPLAT_HMAC_SHA384_CTX_HANDLE;
+EVP_MAC_CTX *CXPLAT_HMAC_SHA512_CTX_HANDLE;
+
+int
+CxPlatLoadCipher(
+    char *cipher_name,
+    EVP_CIPHER **cipher
+    )
+{
+    *cipher = EVP_CIPHER_fetch(NULL, cipher_name, "");
+    if (*cipher == NULL) {
+        QuicTraceEvent(
+            LibraryErrorStatus,
+            "[ lib] ERROR, %u, %s.",
+            ERR_get_error(),
+            cipher_name);
+        return 0;
+    }
+    return 1;
+}
+
+int
+CxPlatLoadMAC(
+    char *name,
+    EVP_MAC **mac
+    )
+{
+    *mac = EVP_MAC_fetch(NULL, name, "");
+    if (*mac == NULL) {
+        QuicTraceEvent(
+            LibraryErrorStatus,
+            "[ lib] ERROR, %u, %s.",
+            ERR_get_error(),
+            "EVP_MAC_fetch failed");
+        return 0;
+    }
+    return 1;
+}
+
+int
+CxPlatLoadHMACCTX(
+    EVP_MAC *mac,
```

Comment:
Are you familiar with SAL annotations?  If so, would you add the appropriate annotations to your new functions?

Reply:
No, I'm not familiar with SAL annotations...

Reply:
I've made a stab at adding the annotations.